### PR TITLE
Add ability to store result variables on service tasks as transient variables

### PR DESCRIPTION
**Pull request by Stoeoeoe**
Added the ability to store the result of expressions as a transient variable in BPMN and CMMN.

### CODE REVIEW

File: modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/behavior/impl/PlanItemExpressionActivityBehavior.java [hunk 1]
```diff
@@ -42,7 +44,11 @@ public void execute(CommandContext commandContext, PlanItemInstanceEntity planIt
             Expression expressionObject = CommandContextUtil.getCmmnEngineConfiguration(commandContext).getExpressionManager().createExpression(expression);
             value = expressionObject.getValue(planItemInstanceEntity);
             if (resultVariable != null) {
-                planItemInstanceEntity.setVariable(resultVariable, value);
+                if(storeResultVariableAsTransient) {
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/behavior/impl/PlanItemExpressionActivityBehavior.java]: missing space "if ("

File: modules/flowable-cmmn-engine/src/test/java/org/flowable/cmmn/test/runtime/ServiceTaskTest.java [hunk 2]
```diff
@@ -173,5 +174,19 @@ public void testDelegateExpressionFields() {
...
+    public void testStoreTransientVariable() {
+        CaseInstance caseInstance = cmmnRuntimeService.createCaseInstanceBuilder()
+            .caseDefinitionKey("myCase")
+            .start();
+
+        Object transientResult = cmmnRuntimeService.getVariable(caseInstance.getId(), "transientResult");
+        Object persistentResult = cmmnRuntimeService.getVariable(caseInstance.getId(), "persistentResult");
+
+        assertNull(transientResult);
+        assertEquals(persistentResult, "Result is: test");
```
Comment (dbmalkovsky) [inline modules/flowable-cmmn-engine/src/test/java/org/flowable/cmmn/test/runtime/ServiceTaskTest.java]: Expected constant should be on the first argument

File: modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/behavior/ServiceTaskExpressionActivityBehavior.java [hunk 3]
```diff
@@ -86,10 +88,18 @@ public void execute(DelegateExecution execution) {
 
                 value = expression.getValue(execution);
                 if (resultVariable != null) {
-                    if (useLocalScopeForResultVariable) {
-                        execution.setVariableLocal(resultVariable, value);
+                    if(storeResultVariableAsTransient) {
```
Comment (dbmalkovsky) [inline modules/flowable-engine/src/main/java/org/flowable/engine/impl/bpmn/behavior/ServiceTaskExpressionActivityBehavior.java]: missing space: "if ("

File: modules/flowable-engine/src/test/java/org/flowable/engine/test/bpmn/servicetask/ServiceTaskTransientVariablesTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    @Test
+    @Deployment
+    public void testStoreTransientVariable() {
+        ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("process");
+
+        Object transientResult = runtimeService.getVariable(processInstance.getId(), "transientResult");
+        Object persistentResult = runtimeService.getVariable(processInstance.getId(), "persistentResult");
+
+        assertNull(transientResult);
+        assertEquals(persistentResult, "Result is: test");
```
Comment (dbmalkovsky) [inline modules/flowable-engine/src/test/java/org/flowable/engine/test/bpmn/servicetask/ServiceTaskTransientVariablesTest.java]: expected value should be the first argument

File: modules/flowable-engine/src/test/java/org/flowable/engine/test/bpmn/servicetask/ServiceTaskTransientVariablesTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Test
+    @Deployment
+    public void testStoreLocalTransientVariable() {
+        ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("process");
+
+        List<HistoricVariableInstance> variablesInstances = historyService.createHistoricVariableInstanceQuery().processInstanceId(processInstance.getId()).list();
+        assertEquals(variablesInstances.size(), 1);
+        assertEquals(variablesInstances.get(0).getValue(), "Result is: test");
```
Comment (dbmalkovsky) [inline modules/flowable-engine/src/test/java/org/flowable/engine/test/bpmn/servicetask/ServiceTaskTransientVariablesTest.java]: lines 48 and 49: expected value should be the first argument

**Comment by tijsrademakers** [comment]
Thanks @Stoeoeoe , could you have a look at the points raised by David and resolve the conflicts in the PR. Then we can merge it to the master branch.

File: modules/flowable-cmmn-json-converter/src/main/java/org/flowable/cmmn/editor/json/converter/ServiceTaskJsonConverter.java [hunk 6]
```diff
@@ -142,6 +142,10 @@ protected BaseElement convertJsonToElement(JsonNode elementNode, JsonNode modelN
             task.setResultVariableName(CmmnJsonConverterUtil.getPropertyValueAsString(PROPERTY_SERVICETASK_RESULT_VARIABLE, elementNode));
         }
 
+        if (CmmnJsonConverterUtil.getPropertyValueAsBoolean(PROPERTY_SERVICETASK_STORE_RESULT_AS_TRANSIENT, elementNode)) {
```
Comment (filiphr) [inline modules/flowable-cmmn-json-converter/src/main/java/org/flowable/cmmn/editor/json/converter/ServiceTaskJsonConverter.java]: Isn't the reverse missing? When importing from the BPMN XML into the Modeler JSON?

Comment (Stoeoeoe) [reply modules/flowable-cmmn-json-converter/src/main/java/org/flowable/cmmn/editor/json/converter/ServiceTaskJsonConverter.java]: You're right, that slipped through the cracks. I added it.

**Comment by filiphr** [comment]
Closing this in favour of PR #2216

**Pull request closed without merging**